morphuntion/src/icu4cxx/BreakIterator.hpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2016-2024 Apple Inc. All rights reserved.
+ */
+// This is a generated file. Do not edit this file.
+#pragma once
+#include <string_view>
+#include <morphuntion/exception/ICUException.hpp>
+#include <morphuntion/exception/NullPointerException.hpp>
+#include <unicode/ubrk.h>
+
+namespace icu4cxx {
+class BreakIterator {
+public:
+BreakIterator(UBreakIteratorType type, const char * locale, std::u16string_view text) {
+    UErrorCode ec = U_ZERO_ERROR;
+    wrappee_ = ubrk_open(type, locale, (const UChar *)text.data(), (int32_t)text.size(), &ec);
+    ::morphuntion::exception::ICUException::throwOnFailure(ec);
```

Comment:
Some companies don't allow exceptions in code. Do these bubble up to top or are they always caught and then API returns an error?

Reply:
They're not caught on the C++ API. The C API wrappers around the C++ code converts the exceptions to CFErrorRef.  If you don't want C++ exceptions, use the C API.  If you want C++ exceptions, use the C++ API.  C++ library code also generates exceptions. So you can't completely omit them from C++.